data.tf

```diff
@@ -37,7 +37,7 @@ data "aws_subnet_ids" "private" {
 
 data "aws_route_table" "this" {
   provider = aws.satellite
-  count    = local.create ? length(data.aws_subnet_ids.this[0].ids) : 0
+  count    = local.create ? length(data.aws_subnet_ids.private[0].ids) : 0
```

Comment:
Nice catch!
Even while looking at the fix now, it wasn't entirely obvious why this was causing an issue.

I think the confusion generated by the data source name [here](https://github.com/Flaconi/terraform-aws-transit-gateway-satellite/pull/16/files#diff-d27b6bbb73d8ee24ac0b6cae34cdddcde2c220a0ed46e41cfd5b36e4a61b41abL10) is not helping at all: `data.aws_subnet_ids.this`

Given that this data source is only used for [attachment purposes](https://github.com/Flaconi/terraform-aws-transit-gateway-satellite/blob/master/main.tf#L7) at this point, my proposal would be to rename it to something like `data.aws_subnet_ids.attachment`. Then, its purpose would be clearly differentiated from `data.aws_subnet_ids.private.`

WDYT @snovikov ?